local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php

```diff
@@ -57,25 +78,42 @@ public function generateUrlFunction($params, &$smarty)
             );
         }
 
+        if ($id && $router) {
+            // get url by router and id
```

Comment:
Nop
when $router or $id = ''
if (null !== $id) return true
and 
if ($id) return false